hud/src/androidLibMain/kotlin/UiContextTrackingBuilder.kt

```diff
@@ -0,0 +1,48 @@
+package com.splendo.kaluga.hud
+
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.LifecycleOwner
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.MainScope
+
+/*
+
+Copyright 2019 Splendo Consulting B.V. The Netherlands
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+
+actual class UiContextTrackingBuilder: CoroutineScope by MainScope() {
+
+    data class UiContextData(
+        val lifecycleOwner: LifecycleOwner,
+        val fragmentManager: FragmentManager
+    )
+
+    internal var onUiContextDataChanged: ((newValue: UiContextData?, oldValue: UiContextData?) -> Unit)? = null
```

Comment:
Probably `protected` instead of `internal`?

Reply:
It says `protected` inside final class the same as `private`, but I have to access it outside this class.

Reply:
Yes I assume this class is not final, you extend it to make a builder right?

Reply:
All classes in Kotlin are final by default :(

Reply:
Yes, you could also consider making this class `abstract`, not sure if it will ever make sense on it's own.